gir/pkgconfig/pkgconfig.go

```diff
@@ -1,105 +1,152 @@
+// Copyright (C) 2021, 2023  diamondburned
+// Copyright (C) 2023  Luke Shumaker
+
 // Package pkgconfig provides a wrapper around the pkg-config binary.
 package pkgconfig
 
 import (
 	"errors"
 	"fmt"
-	"io/fs"
-	"os"
 	"os/exec"
 	"path/filepath"
 	"strings"
-	"sync"
-)
-
-var (
-	path     string
-	pathErr  error
-	pathOnce sync.Once
 )
 
-func ensure() error {
-	pathOnce.Do(func() {
-		path, pathErr = exec.LookPath("pkg-config")
-		if pathErr != nil {
-			pathErr = fmt.Errorf("failed to find pkg-config: %w", pathErr)
-		}
-	})
-	return pathErr
-}
-
-// IncludeDir finds the include directories for the given packages.
-func IncludeDirs(pkgs ...string) ([]string, error) {
-	if err := ensure(); err != nil {
-		return nil, err
+// Variable returns a map of map[pkg]value containing the given
+// variable value for each requested package.  It is not an error for
+// a variable to be unset or empty; ret[pkg] is an empty string if
+// that package does not have the variable set.
+func Variable(varname string, pkgs ...string) (map[string]string, error) {
+	if len(pkgs) == 0 {
+		return nil, nil
 	}
 
-	args := []string{"--variable=includedir"}
-	args = append(args, pkgs...)
-
-	out, err := exec.Command(path, args...).Output()
-	if err != nil {
+	cmdline := append([]string{"pkg-config",
+		// Don't be opaque when we fail.
+		"--print-errors",
+		// On Nix, there are "normal" and "dev" split
+		// packages, with different prefixes.  The "prefix"
+		// variable is set to the normal package's prefix,
+		// while the .pc, .h and .gir files are installed
+		// relative to the dev package's prefix.  So tell
+		// pkg-config to override the prefix variable with the
+		// location of the .pc file.  An alternative way of
+		// getting the dev package's suffix would be to
+		// inspect the "includedirs" variable and trim off the
+		// "/include(/.*)?$" suffix.
+		"--define-prefix",
+		// main
+		"--variable=" + varname, "--"}, pkgs...)
+	var stdout strings.Builder
+	cmd := exec.Command(cmdline[0], cmdline[1:]...)
+	cmd.Stdout = &stdout
+	if err := cmd.Run(); err != nil {
 		var exitErr *exec.ExitError
 		if !errors.As(err, &exitErr) {
 			return nil, fmt.Errorf("pkg-config failed: %w", err)
 		}
-
-		return nil, fmt.Errorf(
-			"pkg-config failed with status %d:\n%s",
-			exitErr.ExitCode(), exitErr.Stderr,
-		)
+		return nil, fmt.Errorf("pkg-config failed with status %d:\n%s",
+			exitErr.ExitCode(), exitErr.Stderr)
 	}
 
-	return strings.Fields(string(out)), nil
-}
+	ret := make(map[string]string, len(pkgs))
+	stdoutStr := strings.TrimRight(stdout.String(), "\n")
+	if stdoutStr == "" {
```

Comment:
When is this the case?

Reply:
With [classic `pkg-config`](https://www.freedesktop.org/wiki/Software/pkg-config/), it'll skip the spaces if the variable is set for *none* of the requested packages (but the new-hotness [pkgconf `pkg-config`](https://gitea.treehouse.systems/ariadne/pkgconf) will still print the spaces).

But, as I double-checked that, I've discovered that pkgconf 2 no longer supports querying multiple packages at once.  If you give it more than one it just silently ignores all but the first. :(

Reply:
I've actually misunderstood what FDO pkg-config is doing; I've now fixed that, added a comment with a link to the FDO source code, and added a test.